pymongo/client_session.py

```diff
@@ -368,6 +377,97 @@ def _inherit_option(self, name, val):
             return val
         return getattr(self.client, name)
 
+    def with_transaction(self, callback, read_concern=None, write_concern=None,
+                         read_preference=None):
+        """Execute a callback in a transaction.
+
+        .. code-block:: python
+
+          def callback(session):
+              orders = session.client.db.orders
+              inventory = session.client.db.inventory
+              orders.insert_one({"sku": "abc123", "qty": 100}, session=session)
+              inventory.update_one({"sku": "abc123", "qty": {"$gte": 100}},
+                                   {"$inc": {"qty": -100}}, session=session)
+
+          with client.start_session() as session:
+              session.with_transaction(callback)
+
+        In an ideal scenario, `with_transaction` will start a transaction,
```

Comment:
The future tense is confusing here. Why not say

```
Ideally, `with_transaction` starts a transaction, executes the callback once, and commits the transaction....
```

Reply:
Updated.